### PR TITLE
docs: add bun runtime explanations

### PR DESCRIPTION
**Pull request by TomChv**
/cc @benjaminstrasser

### CODE REVIEW

File: docs/current_docs/manuals/developer/typescript/356352-runtime.mdx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+Set the field to `bun` to use the Bun runtime.
+
+```json
+  "dagger": {
+    "runtime": "bun"
+  }
+```
+
+:::warning
+The Bun runtime support is still experimental, Dagger cannot guarantee the perfect execution of the function until
```
Comment (vikram-dagger) [inline docs/current_docs/manuals/developer/typescript/356352-runtime.mdx:41]:
```suggestion
Bun runtime support is still experimental. Unexpected results may occur in some cases.
```

File: docs/current_docs/manuals/developer/typescript/356352-runtime.mdx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+Bun is fully compatible with Node.
```
Comment (vikram-dagger) [inline docs/current_docs/manuals/developer/typescript/356352-runtime.mdx:42]:
```suggestion
```

File: docs/current_docs/manuals/developer/typescript/356352-runtime.mdx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+```json
+  "dagger": {
+    "runtime": "bun"
+  }
+```
+
+:::warning
+Bun runtime support is still experimental. Unexpected results may occur in some cases.
+:::
```
Comment (benjaminstrasser) [inline docs/current_docs/manuals/developer/typescript/356352-runtime.mdx:42]:
```suggestion
:::
## Automatic Detection
When a runtime is not explicitly defined within the `package.json` file, Dagger automatically identifies the appropriate runtime by examining the project's lock files inside the projects `dagger` directory.

- If Dagger finds any of the following lock files : `package-lock.json`, `yarn.lock`, or `pnpm-lock.yaml`, it automatically selects `node` as the runtime.

- In the absence of the lock files mentioned above, if a `bun.lockb` file is present, Dagger will choose `bun` as the runtime.

- If no or only unknown lock files are present `node` is choosen.

This behavior however should be seen as a sensible fallback and not as an explicit configuration.
```

We could add something like this at the end. This way potential confusions could be avoided. F.e someone uses bun as a package manager, but still wants to use the Node.js runtime. This section explains daggers behavior in this situation.
I am also open to not include this for the sake of simplicity.

Comment (vikram-dagger) [reply docs/current_docs/manuals/developer/typescript/356352-runtime.mdx:42]: This seems like valuable information to add.

File: docs/current_docs/manuals/developer/typescript/356352-runtime.mdx [hunk 4]
```diff
@@ -0,0 +1,42 @@
+---
+slug: /manuals/developer/typescript/356352/runtime
+displayed_sidebar: "current"
+toc_max_heading_level: 2
+title: "Runtimes"
+---
+
+# Runtimes
+
+Multiple runtimes support TypeScript such as [Node.js](https://nodejs.org/en), [Deno](https://deno.com/) and [Bun](https://bun.sh/).
```
[the remark this answers is not in the document]
Comment (vikram-dagger) [reply docs/current_docs/manuals/developer/typescript/356352-runtime.mdx:10]: I think we should correct this. @TomChv?

[the remark this answers is not in the document]
Comment (TomChv) [reply docs/current_docs/manuals/developer/typescript/356352-runtime.mdx:10]: `tsx` is based on Node.JS runtime no?

[the remark this answers is not in the document]
Comment (benjaminstrasser) [reply docs/current_docs/manuals/developer/typescript/356352-runtime.mdx:10]: @TomChv 
Yes it is. I think the current statement conveys the right message, even though Node does not support typescript. It is just that we use tsx. tsx just handles all the ts compalition(I think using esbuild) and then runs the code using node. Might just be something we would like to add here.

File: docs/current_docs/manuals/developer/typescript/356352-runtime.mdx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+:::
+## Automatic Detection
+When a runtime is not explicitly defined within the `package.json` file, Dagger automatically identifies the appropriate runtime by examining the project's lock files inside the project's `dagger` directory.
+
+- If Dagger finds any of the following lock files : `package-lock.json`, `yarn.lock`, or `pnpm-lock.yaml`, it automatically selects `node` as the runtime.
+- In the absence of the lock files mentioned above, if a `bun.lockb` file is present, Dagger will choose `bun` as the runtime.
+
+- If no or only unknown lock files are present, `node` is chosen.
+
+This behavior however should be seen as a sensible fallback and not as an explicit configuration.
```
Comment (benjaminstrasser) [inline docs/current_docs/manuals/developer/typescript/356352-runtime.mdx:51]:
```suggestion
:::warning
This behavior however should be seen as a sensible fallback and not as an explicit configuration.
:::
```

Maybe add this as a warning?

File: docs/current_docs/manuals/developer/typescript/356352-runtime.mdx (hunk 4 above)
[the remark this answers is not in the document]
Comment (vikram-dagger) [reply docs/current_docs/manuals/developer/typescript/356352-runtime.mdx:10]: Maybe reword it to 

TypeScript is supported by multiple runtimes: [Node.js](https://nodejs.org/en) (via `tsx`), [Deno](https://deno.com/) (natively) and [Bun](https://bun.sh/) (natively).

File: docs/current_docs/manuals/developer/typescript/356352-runtime.mdx [hunk 6]
```diff
@@ -0,0 +1,53 @@
+---
+slug: /manuals/developer/typescript/356352/runtime
+displayed_sidebar: "current"
+toc_max_heading_level: 2
+title: "Runtimes"
+---
+
+# Runtimes
+
+Multiple runtimes support TypeScript such as [Node.js](https://nodejs.org/en), [Deno](https://deno.com/) and [Bun](https://bun.sh/).
```
Comment (vikram-dagger) [inline docs/current_docs/manuals/developer/typescript/356352-runtime.mdx:10]:
```suggestion
TypeScript is supported by multiple runtimes: [Node.js](https://nodejs.org/en) (via tsx), [Deno](https://deno.com/) (natively) and [Bun](https://bun.sh/) (natively).
```

File: docs/current_docs/manuals/developer/typescript/356352-runtime.mdx (hunk 4 above)
[the remark this answers is not in the document]
Comment (vikram-dagger) [reply docs/current_docs/manuals/developer/typescript/356352-runtime.mdx:10]: Added as suggestion

**Review by vikram-dagger — APPROVED** [review]
